src/screens/SettingsModal.js

```diff
@@ -99,45 +118,124 @@ const onPressHiddenFeature = async () => {
   Alert.alert('DEBUG INFO', JSON.stringify(keys, null, 2));
 };
 
+const transitionConfig = {
+  damping: 35,
+  mass: 1,
+  overshootClamping: false,
+  restDisplacementThreshold: 0.01,
+  restSpeedThreshold: 0.01,
+  stiffness: 450,
+};
+
 export default function SettingsModal() {
   const { goBack, navigate } = useNavigation();
-  const [currentSettingsPage, setCurrentSettingsPage] = useState(
-    SettingsPages.default
-  );
+  const { wallets, selectedWallet } = useWallets();
+  const { params } = useRoute();
+  const { isTinyPhone, width: deviceWidth } = useDimensions();
 
-  const { title } = currentSettingsPage;
-  const isDefaultPage = title === SettingsPages.default.title;
-
-  const onPressBack = useCallback(() => {
-    setCurrentSettingsPage(SettingsPages.default);
-    navigate('SettingsSection');
-  }, [navigate, setCurrentSettingsPage]);
+  const getRealRoute = useCallback(
+    key => {
+      let route = key;
+      let params = {};
+      if (key === SettingsPages.backup.key) {
+        const walletId = params?.walletId;
```

Comment:
Since we are defining params in 139 (no longer using the useRoute params), line 141 would always be falsey no?

Reply:
Although, now that I'm looking, I don't see a setParams with walletId anywhere?

Reply:
params on the line 133 is read only (to read the params of the current route).

The params on line  139 is setting the params needed to be passed as parameters to navigate.
I know the naming sucks so I'm renaming it to paramsToPass